# Total planning attempts
max_solutions: 10

# Planning group and link names
arm_group_name: "panda_arm"
eef_name: "hand"
hand_group_name: "hand"
hand_frame: "panda_link8"

# Poses
hand_open_pose: "open"
hand_close_pose: "close"
arm_home_pose: "ready"

# Scene frames
world_frame: "world"
table_reference_frame: "world"
object_reference_frame: "world"
camera_reference_frame: "world"
surface_link: "table"

# Collision object for picking
# CYLINDER object specifications
spawn_mesh: false
object_name: "object"
object_dimensions: [0.25, 0.02] # [height, radius]
object_pose: [0.5, -0.25, 0.0, 0, 0, 0]

# # BAR CLAMP
# spawn_mesh: true
# object_name: "object"
# object_mesh_file: "package://deep_grasp_task/meshes/objects/bar_clamp.dae"
# object_dimensions: [0.06, 0.12, 0.04]  # [length, width, height]
# object_pose: [0.5, 0, 0.03, 0, 0, 0]

# STRAWBERRY
# spawn_mesh: true
# object_name: "object"
# object_mesh_file: "package://deep_grasp_task/meshes/objects/strawberry.dae"
# object_dimensions: [0.02]  # [radius]
# object_pose: [0.5, 0, 0., 0, 0, 0]

# Camera collision object
spawn_camera: false
camera_name: "camera"
camera_mesh_file: "package://deep_grasp_task/meshes/camera/kinect.dae"
camera_pose: [0.5, 0, 0.7, 0, 0, 0] # side
# camera_pose: [0.5, 0, 0.7, 0, 1.571, 1.571] # over head

# Table model
spawn_table: true
table_name: "table"
table_dimensions: [0.4, 0.5, 0.1]  # [length, width, height]
table_pose: [0.5, -0.25, 0, 0, 0, 0]

# Gripper grasp frame transform [x,y,z,r,p,y]
grasp_frame_transform: [0, 0, 0.1, 1.571, 0.785, 1.571]

# Place pose [x,y,z,r,p,y]
# place_pose: [0.5, 0.3, 0, 0, 0, 0]
place_pose: [0.6, -0.15, 0, 0, 0, 0]
place_surface_offset: 0.0001 # place offset from table

# Approach vector in the hand frame
approach_vector_x: 0.0
approach_vector_y: 0.0
approach_vector_z: 1.0

# Valid distance range when approaching an object for picking
approach_object_min_dist: 0.1
approach_object_max_dist: 0.15

# Valid height range when lifting an object after pick
lift_object_min_dist: 0.01
lift_object_max_dist: 0.1
